File: src/main/java/me/nasukhov/TukitaLearner/study/Question.java
package me.nasukhov.TukitaLearner.study;

import jakarta.persistence.*;
import me.nasukhov.TukitaLearner.db.StringListConverter;
import org.springframework.data.annotation.Immutable;

import java.util.List;

@Entity
@Immutable
@Table(name = "questions")
public class Question {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    public String text;

    @Column(nullable = false)
    public String answer;

    @Convert(converter = StringListConverter.class)
    @Column(columnDefinition = "TEXT")
    public List<String> variants;

    @OneToMany(mappedBy = "question", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<GroupQuestion> groupQuestions;

    protected Question() {

    }

    public Question(
            String text,
            String answer,
            List<String> variants
    ) {
        this.text = text;
        this.answer = answer;
        this.variants = variants;
    }
}

File: src/main/kotlin/me/nasukhov/tukitalearner/study/Question.kt
package me.nasukhov.tukitalearner.study

import jakarta.persistence.*
import me.nasukhov.utility.db.StringListConverter
import org.springframework.data.annotation.Immutable

@Entity
@Immutable
@Table(name = "questions")
class Question {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private var id: Long = 0

    @Column(nullable = false)
    lateinit var text: String
        private set

    @Column(nullable = false)
    lateinit var answer: String
        private set

    @Convert(converter = StringListConverter::class)
    @Column(columnDefinition = "TEXT")
    lateinit var variants: List<String>
        private set

    private constructor()

    constructor(
        text: String,
        answer: String,
        variants: List<String>,
    ) {
        this.text = text
        this.answer = answer
        this.variants = variants
    }
}
